odyssey-app/src/main/java/com/codebutler/odyssey/app/feature/game/Square.kt

```diff
@@ -94,37 +124,74 @@ class Square {
 
     fun draw() {
         val b = bitmap ?: return
-        val textureLocation = GLRenderer2d.loadTexture(b)
 
         // Add program to OpenGL ES environment
-        GLES20.glUseProgram(program)
+        glUseProgram(program)
+
+        // bind the texture
+        val textureLocation = bindTexture(b)
+        glActiveTexture(0)
+        glBindTexture(GL_TEXTURE_2D, textureLocation)
+        glUniform1i(glGetUniformLocation(program, "u_Texture"), 0)
```

Comment:
I was trying to figure out how you were actually drawing the texture, since you never bound it in the shader.  It looks like it was *accidentally* being bound, since the `sampler2D` defaults to 0 and the texture was created on the current active texture of 0.  I made it explicit here so that doesn't get lost in the future if we want to add any other textures to the screen.

Reply:
Lol yup.